server/rpc/handle_get_block_template.go

```diff
@@ -784,7 +787,9 @@ func (state *gbtWorkState) blockTemplateResult(dag *blockdag.BlockDAG, useCoinba
 	// implied by the included or omission of fields:
 	//  Including MinTime -> time/decrement
 	//  Omitting CoinbaseTxn -> coinbase, generation
-	targetDifficulty := fmt.Sprintf("%064x", util.CompactToBig(header.Bits))
+	target := big.NewInt(0)
+	util.CompactToBig(header.Bits, target)
```

Comment:
Maybe create function that wraps CompactToBig that returns the result, in addition to the one with the out param.

You optimized one place, but all other places have become a bit more verbose because of it. Better be able to enjoy both worlds.